iothub/service/src/DigitalTwin/DigitalTwinClient.cs

```diff
@@ -36,7 +40,39 @@ public static DigitalTwinClient CreateFromConnectionString(string connectionStri
 
         private DigitalTwinClient(Uri uri, IotServiceClientCredentials credentials, params DelegatingHandler[] handlers)
         {
-            _client = new IotHubGatewayServiceAPIs(uri, credentials, handlers);
+            // ConnectionLeaseTimeoutHandler is IDisposable, so save it as a field so it can be disposed later.
+            _connectionLeaseTimeoutHandler = new ConnectionLeaseTimeoutHandler(HttpTransportSettings.DefaultConnectionLeaseTimeoutMilliseconds);
+
+            // Assign each custom delegating handler as the inner handler of the previous delegating handler
+            // For instance:
+            // _connectionLeaseTimeoutHandler.InnerHandler = handlers[0],
+            // handlers[0].InnerHandler = handlers[1],
+            // handlers[1].InnerHandler = handlers[2],
+            // ...
+            // handlers[handlers.Length].InnerHandler = _httpClientHandler
+            DelegatingHandler previousDelegatingHandler = _connectionLeaseTimeoutHandler;
+            foreach (DelegatingHandler delegatingHandler in handlers)
+            {
+                previousDelegatingHandler.InnerHandler = delegatingHandler;
+                previousDelegatingHandler = delegatingHandler;
+            }
+
+            _httpClientHandler = new HttpClientHandler()
+            {
+#if !NET451
+                SslProtocols = TlsVersions.Instance.Preferred,
+                CheckCertificateRevocationList = TlsVersions.Instance.CertificateRevocationCheck,
+#endif
+            };
+
+            previousDelegatingHandler.InnerHandler = _httpClientHandler;
```

Comment:
This looks good! Some (nit) comments:
- To me, `previousDelegatingHandler` doesn't really share the whole context of the "pipeline" process. Maybe could be renamed to something like `chainedDelegatingHandler`?
- Add a note stating that the `HttpClientHandler` is the one making the network requests, so it needs to be at the end of the pipeline.